File: mobile/lib/solver/cube_state_draft.dart
import 'solver_models.dart';

/// Mirrors CubeStateDraft.swift's role in the superseded native app: the
/// Application Layer's own thin wrapper around "a cube state the user
/// picked", carrying a human-readable label alongside the SolverCubie list
/// so the Solve screen doesn't need SolverFixtureCubies in scope.
class CubeStateDraft {
  final List<SolverCubie> cubies;
  final String scrambleLabel;

  const CubeStateDraft({required this.cubies, required this.scrambleLabel});
}

File: ios/PolyPuzzleCube/Features/CubeInput/CubeStateDraft.swift
import Foundation
import PolyPuzzleSolverSDK

/// The Application Layer's own representation of "a cube the user is
/// looking at right now" -- deliberately thin: it wraps exactly the
/// [SolverCubie] the Solver SDK needs, plus a human-readable summary for
/// the Home/CubeInput screens. Kept separate from SolverRequest itself so
/// UI code never has to think about `weights`/`endgameReserveMs` (SDK-level
/// tuning knobs the directive's minimum feature set doesn't expose).
struct CubeStateDraft: Hashable {
  var cubies: [SolverCubie]
  var scrambleLabel: String

  static func solved() -> CubeStateDraft {
    CubeStateDraft(cubies: SolverFixtureCubies.solved(), scrambleLabel: "Solved")
  }
}
